tests/test_config.py

```diff
@@ -883,3 +883,76 @@ def test_alias_volumes_set(self):
         assert v.container_path == '/bar'
         assert v.host_path == '/host/bar'
         assert v.options == ['z', 'ro']
+
+    def test_volumes_with_env_vars_simple(self):
+        '''volume definitions can contain environmental variables'''
+        os.environ["TEST_VOL_PATH"] = "/bar/baz"
+        os.environ["TEST_VOL_PATH2"] = "/moo/doo"
+        with open('.scuba.yml', 'w') as f:
+            f.write(r'''
+                image: na
+                volumes:
+                  $TEST_VOL_PATH/foo: ${TEST_VOL_PATH2}/foo
+                ''')
+
+        config = scuba.config.load_config('.scuba.yml')
+        vols = config.volumes
+        assert len(vols) == 1
+
+        v = list(vols.values())[0]
+        assert isinstance(v, scuba.config.ScubaVolume)
+        assert v.container_path == '/bar/baz/foo'
+        assert v.host_path == '/moo/doo/foo'
+        assert v.options == []
+
+    def test_volumes_with_env_vars_complex(self):
+        '''complex volume definitions can contain environmental variables'''
+        os.environ["TEST_HOME"] = "/home/testuser"
+        os.environ["TEST_TMP"] = "/tmp"
+        os.environ["TEST_MAIL"] = "/var/spool/mail/testuser"
```

Comment:
This actually modifies the process-global environment, which is not a good practice for unit tests, and can lead to confusing bugs.

The right way is to _mock_ the environment. Luckily we're using `pytest` which provides [`monkeypatch.setenv()`](https://docs.pytest.org/en/6.2.x/monkeypatch.html). See [this question](https://stackoverflow.com/q/38410878/119527).

This is already being used in `test_utils.py`:
https://github.com/JonathonReinhart/scuba/blob/04dc2c2c77de136bbe2caee8a1cbf2909564a610/tests/test_utils.py#L69

So after adding the fixture (by simply adding the parameter named `monkeypatch`):
```python
    def test_volumes_with_env_vars_complex(self, monkeypatch):
```
You can then:
```suggestion
        monkeypatch.setenv("TEST_HOME", "/home/testuser")
        monkeypatch.setenv("TEST_TMP", "/tmp")
        monkeypatch.setenv("TEST_MAIL", "/var/spool/mail/testuser")
```

Reply:
In an earlier version of these tests I was modifying `$HOME`, which caused later tests to fail (but only because of the order the tests are executed in). I've been away from the world of high-level languages (with great tooling) too long, so I forgot about all of the great Python mocking features that exist to facilitate testing!

Reply:
Test now mock the environment in 2405ed8